sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -888,19 +934,24 @@ template <atomic_op Op, typename Tx, int N, typename Toffset>
 __ESIMD_API simd<Tx, N> atomic_update(Tx *p, simd<Toffset, N> offset,
                                       simd_mask<N> mask) {
   static_assert(std::is_integral_v<Toffset>, "Unsupported offset type");
-  detail::check_atomic<Op, Tx, N, 0>();
-  simd<uintptr_t, N> vAddr(reinterpret_cast<uintptr_t>(p));
-  simd<uintptr_t, N> offset_i1 = convert<uintptr_t>(offset);
-  vAddr += offset_i1;
-  using T = typename detail::__raw_t<Tx>;
-  return __esimd_svm_atomic0<Op, T, N>(vAddr.data(), mask.data());
+  if constexpr (Op == atomic_op::load) {
```

Comment:
Please add static_assert() checking that op can be 'load', 'inc' or 'dec'

Reply:
Ough..., right. Sorry for confusion. I missed the fact that check_atomic verifies the num of operands vs the operation.

This makes this static_assert here and on Lin 1036 excessive/redundant.
Please keep the call of check_atomic though, which is still needed.

Reply:
Thank you!